dbt/adapters/bigquery/impl.py

```diff
@@ -115,6 +115,7 @@ class BigqueryConfig(AdapterConfig):
     max_staleness: Optional[str] = None
```

Comment:
do we need a post_init check to enforce that this is required if `allow_non_incremental_definition` is also specified?

Reply:
I could add validation rules if desired. There wasn't really an interest previously in doing so, but there's a small framework for running validation rules already, so I can tap into that.